quinn-udp/src/cmsg/unix.rs

```diff
@@ -43,7 +43,15 @@ impl MsgHdr for crate::imp::msghdr_x {
 
     fn cmsg_nxt_hdr(&self, cmsg: &Self::ControlMessage) -> *mut Self::ControlMessage {
         let selfp = self as *const _ as *mut libc::msghdr;
-        unsafe { libc::CMSG_NXTHDR(selfp, cmsg) }
+        let next = unsafe { libc::CMSG_NXTHDR(selfp, cmsg) };
+
+        if next.is_null()
+            || unsafe { (*next).cmsg_len as usize } < std::mem::size_of::<libc::cmsghdr>()
+        {
+            return std::ptr::null_mut();
+        }
```

Comment:
Maybe leave a comment here about why we're doing this?

Reply:
:+1: Done. Unfortunately I was not able to find an official bug report for this.